src/hooks/demoMode.hook.js

```diff
@@ -1,15 +1,37 @@
-import { useLocation } from 'react-router-dom'
 import { useLayoutEffect, useState } from 'react'
+import localStorageService from '../utils/localStorageService'
+import { isURLMode } from '../utils/helper'
 
-import { isDemoMode } from '../utils/helper'
+/**
+ * @hook
+ * Returns an appliction `mode` object
+ *
+ * isDemoMode = Not ready for production or testing (Missing UI or BE). shows all the code for staging + demo
+ *
+ * isStagingMode = When ready for testing but not for production. Specific for staging
+ *
+ * @returns {Object} Object
+ *
+ * @example
+ *
+ * { isDemoMode: true | false, isStagingMode: true | false}
+ */
 
-export const useDemoMode = () => {
-  const [isDemo, setIsDemo] = useState(false)
-  const location = useLocation()
+export const useMode = () => {
+  const [mode, setMode] = useState(localStorageService.getStorageValue('mode'))
+  const urlMode = isURLMode(window.location.search)
```

Comment:
As I understand `isURLMode` returns the string and not the boolean. So the name must start with `get`. Also acronyms we write not as uppercase but as capitalized values. So, it should be `getUrlMode`